dora/core/client/fs/src/main/java/alluxio/client/DoraClientFileSystemManager.java

```diff
@@ -0,0 +1,174 @@
+          conf.getClass(PropertyKey.USER_CLIENT_OVERRIDE_MODULE_CLASS);
+      try {
+        AbstractModule overrideModuleInstance =
+            overrideModule.getDeclaredConstructor().newInstance();
+        injector = Guice.createInjector(Modules.override(modules).with(overrideModuleInstance));
+      } catch (InstantiationException | IllegalAccessException
+               | InvocationTargetException | NoSuchMethodException e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      injector = Guice.createInjector(modules);
+    }
+    return injector.getInstance(DoraClientFileSystemManager.class);
+  }
+
+  /**
+   * Constructor.
+   * @param fileSystemOptions
+   * @param doraCacheClientFactory
+   * @param mountTableManagerOpt
+   * @param cacheManagerOpt
+   */
+  @Inject
+  public DoraClientFileSystemManager(
+      FileSystemOptions fileSystemOptions,
+      DoraCacheClientFactory doraCacheClientFactory,
+      Optional<MountTableManager> mountTableManagerOpt,
+      Optional<CacheManager> cacheManagerOpt) {
+    mFileSystemOptions = requireNonNull(fileSystemOptions, "fileSystemOptions is null");
+    mDoraClientFactory = requireNonNull(doraCacheClientFactory,
+        "doraCacheClientFactory is null");
+    mMountTableManagerOpt = requireNonNull(mountTableManagerOpt,
+        "mountTableManagerOpt is null");
+    mCacheManagerOpt = requireNonNull(cacheManagerOpt, "cacheManagerOpt is null");
+  }
+
+  /**
+   * @param context
+   * @return a new FileSystem instance
+   */
+  public FileSystem create(FileSystemContext context) {
+    return create(context, mFileSystemOptions);
+  }
+
+  /**
+   * @param context the FileSystemContext to use with the FileSystem
+   * @param options FileSystemOptions
+   * @return a new FileSystem instance
+   */
+  public FileSystem create(FileSystemContext context, FileSystemOptions options) {
+    AlluxioConfiguration conf = context.getClusterConf();
+    checkSortConf(conf);
+    Optional<UfsFileSystemOptions> ufsOptions = options.getUfsFileSystemOptions();
+    Preconditions.checkArgument(ufsOptions.isPresent(),
+        "Missing UfsFileSystemOptions in FileSystemOptions");
+    FileSystem fs = new UfsBaseFileSystem(context,
+        options.getUfsFileSystemOptions().get());
+
+    if (options.isDoraCacheEnabled()) {
+      LOG.debug("Dora cache enabled");
+      fs = new DoraCacheFileSystem(fs, context,
+          mDoraClientFactory.create(context), mMountTableManagerOpt);
+    }
+    if (options.isMetadataCacheEnabled()) {
+      LOG.debug("Client metadata caching enabled");
+      fs = new MetadataCachingFileSystem(fs, context);
+    }
+    if (options.isDataCacheEnabled()
+        && CommonUtils.PROCESS_TYPE.get() == CommonUtils.ProcessType.CLIENT
+        && mCacheManagerOpt.isPresent()
+    ) {
+      LOG.debug("Client local data caching enabled");
+      return new LocalCacheFileSystem(mCacheManagerOpt.get(), fs, conf);
+    }
```

Comment:
so are we still calling `new LocalCacheFileSystem(new MetadataCachingFileSystem(new DoraCacheFileSystem()))`? We didn't really get rid of the nested structure?